src/Dont/DeSerialize.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont;
+
+use Dont\Exception\NonDeSerializableObject;
+use Dont\Exception\TypeError;
+
+trait DeSerialize
```

Comment:
As discussed, `deserialize` would be proper English, while PHP has its own `unserialize` wording. Unsure about the wording to adopt for now. @asgrim, can you give us an opinion?